modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/MimeTypeParametersDeclarationEnricher.java

```diff
@@ -211,24 +174,33 @@ public void visitBinaryType(BinaryType binaryType) {
 
           replaceOutputType(declaration, property, format -> {
             BinaryTypeBuilder builder = BaseTypeBuilder.create(format).binaryType();
-            declaration.getOutput().getType().getAnnotation(ClassInformationAnnotation.class).ifPresent(builder::with);
+            enrichWithAnnotations(builder, declaration.getOutput().getType().getAnnotations());
 
             return builder.build();
           });
         }
+
+        private void enrichWithAnnotations(WithAnnotation withAnnotationBuilder, Set<TypeAnnotation> annotations) {
+          annotations.forEach(typeAnnotation -> withAnnotationBuilder.with(typeAnnotation));
+        }
       });
     }
 
     private void replaceOutputType(ExecutableComponentDeclaration<?> declaration, MediaTypeModelProperty property,
                                    Function<MetadataFormat, MetadataType> type) {
-
-      final MediaType mediaType = property.getMediaType();
-      if (mediaType.matches(ANY)) {
+      if (!shouldOverrideMetadataFormat(declaration)) {
         return;
       }
 
-      final OutputDeclaration output = declaration.getOutput();
-      output.setType(type.apply(toMetadataFormat(mediaType)), output.hasDynamicType());
+      property.getMediaType().ifPresent(mediaType -> {
+        final OutputDeclaration output = declaration.getOutput();
+        output.setType(type.apply(toMetadataFormat(mediaType)), output.hasDynamicType());
+      });
+    }
+
+    private boolean shouldOverrideMetadataFormat(ExecutableComponentDeclaration declaration) {
```

Comment:
explain the inner logic, seems to be random rules, but surely makes sense for something